src/providers/wms/qgswmsnewconnection.cpp

```diff
@@ -0,0 +1,140 @@
+/***************************************************************************
+  qgswmsnewconnection.cpp - QgsWmsNewConnection
+
+ ---------------------
+ begin                : 16.10.2025
+ copyright            : (C) 2025 by Alessandro Pasotti
+ email                : elpaso at itopen dot it
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+#include "qgswmsnewconnection.h"
+#include "qgsowsconnection.h"
+#include "qgswmsprovider.h"
+#include "moc_qgswmsnewconnection.cpp"
+#include "qgsproject.h"
+
+#include <QMessageBox>
+
+QgsWmsNewConnection::QgsWmsNewConnection( QWidget *parent, const QString &connName )
+  : QgsNewHttpConnection( parent, QgsNewHttpConnection::ConnectionWms, QStringLiteral( "WMS" ), connName )
+{
+  connect( wmsFormatDetectButton(), &QPushButton::clicked, this, &QgsWmsNewConnection::detectFormat );
+  initWmsSpecificSettings();
+}
+
+QgsWmsNewConnection::~QgsWmsNewConnection()
+{
+  if ( mCapabilities )
+  {
+    QApplication::restoreOverrideCursor();
```

Comment:
This is messy - use a QgsTemporaryCursorOverride in a std::unique_ptr member variable instead.